### PR TITLE
Bazelise the repository

### PR DESCRIPTION
**Pull request by lolski**
This PR introduces the Bazel build system to the repository, which is intended to make the repo be uniform in structure with other TypeDB repositories.

Additionally, we've reorganised the `datasets` package to allow more datasets to be introduced in the future in a cleaner manner.

### CODE REVIEW

File: BUILD [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# This unpublished material is proprietary to Vaticle.
+# All rights reserved. The methods and
+# techniques described herein are considered trade secrets
+# and/or confidential. Reproduction or distribution, in whole
+# or in part, is forbidden except by express written permission
+# of Vaticle.
```
Comment (krishnangovindraj) [inline BUILD]: The main repository license is AGPL.

File: datasets/bookstore/data.tql
Comment (krishnangovindraj) [inline datasets/bookstore/data.tql:1]: This move will break the existing docker container for everyone. The `bootstrapper.py` script that runs as the docker command reads the `bootstrapper.yml` from the master branch of this repository on bootup.

Please prepend `bookstore` to the paths in `/bootstrapper.yml`, and test by overriding the config. 
`docker run -p 1729:1729 -e BOOTSTRAPPER_CONFIG=<url-of-the-raw-bootstrapper.yml-on-your-branch>`

**Review by krishnangovindraj — CHANGES_REQUESTED** [review]
Needs changes to `bootstrapper.yml` to avoid breaking TypeDB academy.

File: BUILD (hunk 1 above)
Comment (lolski) [reply BUILD]: Fixed in https://github.com/typedb/typedb-sample-data/pull/11/commits/21bacc6bc0c946c5aa09aa29827721e6dbf09f56

File: datasets/bookstore/data.tql
Comment (lolski) [reply datasets/bookstore/data.tql:1]: Fixed in https://github.com/typedb/typedb-sample-data/pull/11/commits/a28e8d9eb15b459d05800ae868eeeac76a4361c7

**Review by krishnangovindraj — APPROVED** [review]
lgtm

**Pull request merged**